### PR TITLE
feat: add `unused` linter

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR adds the `unused` linter to the linter configuration, and removes unused code segments

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: tm2/pkg/amino/genproto/bindings.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func generateMethodsForType(imports *ast.GenDecl, scope *ast.Scope, pkg *amino.P
 		))
 	}
 
-	//////////////////
+	// ////////////////
```
Comment (thehowl) [inline tm2/pkg/amino/genproto/bindings.go:172]: eurgh

can we maybe make this `// -----------` so that even if fmt has to auto-fix it it's not an eyesore?

**Review by thehowl — APPROVED** [review]

File: tm2/pkg/amino/genproto/bindings.go (hunk 1 above)
Comment (zivkovicmilos) [reply tm2/pkg/amino/genproto/bindings.go]: I've tried to do a search and replace for these weird line comments:


4c4a878

File: tm2/pkg/amino/genproto/bindings.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func generateMethodsForType(imports *ast.GenDecl, scope *ast.Scope, pkg *amino.P
 	}
 	dpbote_ := pbote_[1:]
 
-	// ////////////////
+	// // -----------
```
Comment (thehowl) [inline tm2/pkg/amino/genproto/bindings.go:108]: Uhhhhhh

Comment (zivkovicmilos) [reply tm2/pkg/amino/genproto/bindings.go]: 🤷‍♂️ 

Fixed:
[ab38a21](https://github.com/gnolang/gno/pull/1294/commits/ab38a21e7986be1e63243a42677fef97ebd90c14)

**Comment by thehowl** [comment]
pinging @moul for approve&merge

**Review by moul — APPROVED** [review]

**Pull request merged**